runtime/lua/provider/python/health.lua

```diff
@@ -0,0 +1,490 @@
+    vim.list_extend(metas, vim.fn.glob(base .. '.egg-info/PKG-INFO', 1, 1))
+    metas = table.sort(metas, compare)
+
+    if metas and next(metas) ~= nil then
+      for line in io.lines(metas[1]) do
+        local version = line:match('^Version: (%S+)')
+        if version then
+          nvim_version = version
+          break
+        end
+      end
+    end
+  end
+
+  local nvim_path_base = vim.fn.fnamemodify(nvim_path, [[:~:h]])
+  local version_status = 'unknown; ' .. nvim_path_base
+  if is_bad_response(nvim_version) and is_bad_response(pypi_version) then
+    if vim.version.lt(nvim_version, pypi_version) then
+      version_status = 'outdated; from ' .. nvim_path_base
+    else
+      version_status = 'up to date'
+    end
+  end
+
+  return { python_version, nvim_version, pypi_version, version_status }
+end
+
+function M.check()
+  health.start('Python 3 provider (optional)')
+
+  local pyname = 'python3'
+  local python_exe = ''
+  local virtual_env = os.getenv('VIRTUAL_ENV')
+  local venv = virtual_env and vim.fn.resolve(virtual_env) or ''
+  local host_prog_var = pyname .. '_host_prog'
+  local python_multiple = {}
+
+  if health.provider_disabled(pyname) then
+    return
+  end
+
+  local pyenv_table = check_for_pyenv()
+  local pyenv = pyenv_table[1]
+  local pyenv_root = pyenv_table[2]
+
+  if vim.g[host_prog_var] then
+    local message = string.format('Using: g:%s = "%s"', host_prog_var, vim.g[host_prog_var])
+    health.info(message)
+  end
+
+  local python_table = vim.fn['provider#pythonx#Detect'](3)
+  pyname = python_table[1]
+  local pythonx_warnings = python_table[2]
+
+  if is_blank(pyname) then
+    health.warn(
+      'No Python executable found that can `import neovim`. '
+      .. 'Using the first available executable for diagnostics.'
+    )
+  elseif vim.g[host_prog_var] then
+    python_exe = pyname
+  end
+
+  -- No Python executable could `import neovim`, or host_prog_var was used.
+  if not is_blank(pythonx_warnings) then
+    health.warn(pythonx_warnings, {
+      'See :help provider-python for more information.',
+      'You may disable this provider (and warning) by adding `let g:loaded_python3_provider = 0` to your init.vim',
+    })
+  elseif not is_blank(pyname) and is_blank(python_exe) then
+    if not vim.g[host_prog_var] then
+      local message = string.format(
+        '`g:%s` is not set. Searching for %s in the environment.',
+        host_prog_var,
+        pyname
+      )
+      health.info(message)
+    end
+
+    if not is_blank(pyenv) then
+      python_exe = health.system({ pyenv, 'which', pyname }, { stderr = true })
+      if is_blank(python_exe) then
+        health.warn('pyenv could not find ' .. pyname .. '.')
+      end
+    end
+
+    if is_blank(python_exe) then
+      python_exe = vim.fn.exepath(pyname)
+
+      if os.getenv('PATH') then
+        local path_sep = iswin and ';' or ':'
+        local paths = vim.split(os.getenv('PATH') or '', path_sep)
```

Comment:
```suggestion
        local paths = vim.split(os.getenv('PATH'), path_sep)
```

Reply:
Sorry for the second-hand review, but doesn't `os.getenv()` return `nil` if environment variable doesn't exist?

Reply:
Eh, in hindsight never mind, `$PATH` should always be defined. Ignore :)

Reply:
You're smart lol